packages/editor/src/components/template-part-menu-items/convert-to-template-part.js

```diff
@@ -13,7 +13,7 @@ import { symbolFilled } from '@wordpress/icons';
 /**
  * Internal dependencies
  */
-import CreateTemplatePartModal from '../create-template-part-modal';
+import { CreateTemplatePartModal } from '@wordpress/fields';
```

Comment:
Yes, I agree this is weird, but also not sure what's the best place for these wp-core-data aware components. That said I can live with it being in fields for now.

Reply:
I've looked where this modal is being used. It looks like there are two additional places other than the action:

| Patterns > Add new | Block Toolbar | 
| --- | --- |
| <img width="447" alt="Screenshot 2024-12-04 at 11 57 19" src="https://github.com/user-attachments/assets/40082c9f-6b8b-4829-ad15-5f9e1c55bc1c"> | <img width="590" alt="Screenshot 2024-12-04 at 11 58 56" src="https://github.com/user-attachments/assets/4aa344f3-0e68-458e-89b0-c8238de91a81"> |

The second case is a bit more difficult to absorb declaratively because it's very custom. However, it seems the first one ("add new") we should be able to absorb as part of the actions registration. Note how all entities have it. It's a bit complex in patterns due to how they are setup, but I presume we want the same design pattern ("add new" button) for any CPT:

| Patterns | Templates | Pages |
| --- | --- | --- |
| <img width="447" alt="Screenshot 2024-12-04 at 11 57 19" src="https://github.com/user-attachments/assets/40082c9f-6b8b-4829-ad15-5f9e1c55bc1c"> | <img width="507" alt="Screenshot 2024-12-04 at 12 02 12" src="https://github.com/user-attachments/assets/28ad8b8e-488b-426f-9cfb-c9c0890d0af3"> | <img width="507" alt="Screenshot 2024-12-04 at 12 02 03" src="https://github.com/user-attachments/assets/4ba54168-6ef0-424c-bee7-b8c374e8d287"> |

We could look at declare some actions as "insertion", similarly to how some of them are "destructive", perhaps.